### PR TITLE
Do not allow spaces within MediaType's parameters

### PR DESCRIPTION
**Pull request by pgomulka**
Per https://tools.ietf.org/html/rfc7231#section-3.1.1.1 MediaType can
have spaces around parameters pair, but do not allow spaces within
parameter. That means that parameter pair forbids spaces around `=`

follow up after
https://github.com/elastic/elasticsearch/pull/64406#discussion_r517533475
relates #51816

### CODE REVIEW

File: libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java [hunk 1]
```diff
@@ -90,16 +91,14 @@ public static ParsedMediaType parseMediaType(String headerValue) {
...
-                    // https://tools.ietf.org/html/rfc7231#section-3.1.1.1 disallows this
-                    String[] keyValueParam = elements[i].trim().split("=");
-                    if (keyValueParam.length == 2) {
-                        String parameterName = keyValueParam[0].toLowerCase(Locale.ROOT).trim();
-                        String parameterValue = keyValueParam[1].toLowerCase(Locale.ROOT).trim();
-                        parameters.put(parameterName, parameterValue);
-                    } else {
+                    //spaces are allowed between parameters, but not between '=' sign
+                    String[] keyValueParam = paramsAsString.split("=");
+                    if (keyValueParam.length != 2 || hasSpaces(keyValueParam[0]) || hasSpaces(keyValueParam[1])) {
```
Comment (jakelandis) [inline libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java:96]: I think for `keyValueParam[0]` you only want to check trailing whitespace, and for `keyValueParam[1]` you only want to check for leading whitespace. `trim()` checks for both.

Comment (jakelandis) [reply libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java:96]: I think you can use https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isWhitespace(char) for the first and last characters to check them independently.

Comment (pgomulka) [reply libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java:96]: good idea, I was meant to refactor with trim as it was unnecessarily creating garbage

**Review by jakelandis — APPROVED** [review]
LGTM

**Pull request merged**